allennlp/data/fields/knowlege_graph_field.py

```diff
@@ -0,0 +1,58 @@
+"""
+``KnowledgeGraphField`` is a ``Field`` which stores a knowledge graph representation.
+"""
+
+from typing import Dict, List
+
+from overrides import overrides
+
+from allennlp.data.vocabulary import Vocabulary
+from allennlp.data.fields.field import Field, DataArray
+from allennlp.data.token_indexers.token_indexer import TokenIndexer
+from allennlp.data.knowledge_graph import KnowledgeGraph
+
+
+class KnowledgeGraphField(Field[DataArray]):
+    """
+    ``KnowledgeGraphField`` indexes entities and collects information necessary for embedding the knowledge
+    graph. Here we assume the token indexers will index all the information needed to embed entities
+    (say names and types).
+
+    Parameters
+    ----------
+    knowledge_graph : ``KnowledgeGraph``
+        The knowledge graph that this field stores.
+    token_indexers : ``Dict[str, TokenIndexer]``
+        Token indexers for indexing various aspects of entities (say names and type of entities) for embedding
+        them.
+    """
+    def __init__(self,
+                 knowledge_graph: KnowledgeGraph,
+                 token_indexers: Dict[str, TokenIndexer]) -> None:
+        self._knowledge_graph: KnowledgeGraph = knowledge_graph
+        self._token_indexers: Dict[str, TokenIndexer] = token_indexers
+        # {entity: {indexer: indexed_tokens}}
+        self._indexed_entities: Dict[str, Dict[str, List[int]]] = None
+
+    @overrides
+    def index(self, vocab: Vocabulary):
+        entity_arrays = {}
+        for entity in self._knowledge_graph.get_all_entities():
+            indexed_entity = {indexer_name: indexer.token_to_indices(entity, vocab)
+                              for indexer_name, indexer in self._token_indexers.items()}
+            entity_arrays[entity] = indexed_entity
+        self._indexed_entities = entity_arrays
+
+    @overrides
+    def get_padding_lengths(self) -> Dict[str, int]:
+        # pylint: disable=no-self-use
+        return {}
```

Comment:
I'm not confident this is correct.  Can you add some tests for this `Field` to show the intended use?

Reply:
You're right. This is incorrect. I was going to add a TODO to implement this correctly after I have the relevant entity token indexers ready, but forgot. I implemented this method now, but can add tests only after I implement the two entity indexers (name and type), and I plan to do that in the next PR with entity embedding and linking. Is that fine? I added a TODO for this.